v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/FailsafeModJsonToPubsubMessageTransformer.java

```diff
@@ -186,67 +220,78 @@ private PubsubMessage publishModJsonStringToPubsubMessage(String modJsonString)
                                     .build();
                     Topic topic = topicAdminClient.getTopic(request);
```

Comment:
This function `publishModJsonStringToPubsubMessage` is data path with hight rate of throughput, we shouldn't be calling any AdminAPIs here as we'll quickly run out of quota. GetTopicRequest can be made at the launch of the pipeline and later any settings such as messageFormat and messageEncoding should be passed as metadata to steps in the pipeline as things like topic name.